core/consensus/common/primitives/src/lib.rs

```diff
@@ -18,10 +18,23 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use codec::Codec;
+use codec::{Encode, Decode, Codec};
 use client::decl_runtime_apis;
 use rstd::vec::Vec;
 
+/// Represents an Babe equivocation proof.
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(Clone, PartialEq, Eq, Decode, Encode)]
+pub struct EquivocationProof<H, P, S> {
+	pub reporter: P,
+	pub identity: P,
+	pub slot: u64,
```

Comment:
should use a strong type alias as in the one from babe primitives.

I'm not sure why `consensus-common` needs to define this type. Trying to unify things like equivocation proofs across all consensus engines is going to fail.